usaon_benefit_tool/routes/user.py

```diff
@@ -31,7 +31,8 @@ def get(user_id: str):
 @user_bp.route('/<user_id>', methods=['POST'])
 @login_required
 def post(user_id: str):
-    forbid_except_for_roles([RoleName.ADMIN])
+    if user_id != request.form['id']:
```

Comment:
Will these two values always match (left side is the user_id in the URL, and right side is the value in the form)? I think this would be the method to check against the current user's ID:

```suggestion
    if user_id != current_user.id:
```